### PR TITLE
(DEV-4222) Adds Tracking to Prayer

### PR DESCRIPTION
**Pull request by redreceipt**
### What does this PR do, or why is it needed?

Adds tracking to various parts of the prayer section

### How do I test this PR?

¯\\\_(ツ)\_/¯

## TODO

- [x] I am affirming this is my _best_ work ([Ecclesiastes 9:10](https://www.bible.com/bible/97/ECC.9.10.MSG))
- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android if applicable
- [x] Set a relevant reviewer

## REVIEW

- [ ] Review updates to test coverage and snapshots
- [ ] Review code through the lens of being concise, simple, and well-documented

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

---

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/newspringchurchapp/src/prayer/SaveButton/index.js [hunk 1]
```diff
@@ -11,22 +12,27 @@ const SaveButtonConnected = memo(({ prayerID, saved, toggleSavedState }) => (
...
+                    unSave({
+                      variables: { nodeId: prayerID },
+                      refetchQueries: [{ query: GET_SAVED_PRAYERS }],
+                    });
+                  else
+                    save({
+                      variables: { nodeId: prayerID },
+                      refetchQueries: [{ query: GET_SAVED_PRAYERS }],
+                    });
+                  track({ eventName: saved ? 'Unsave' : 'Save' });
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/SaveButton/index.js]: This just feels clearer.
```suggestion
                  track({ eventName: saved ? 'Unsaved Prayer' : 'Saved Prayer' });
```

**Comment by richarddubay** [comment]
Can we get the data on when someone taps on a prayer category and which category that was?

**Comment by redreceipt** [comment]
> Can we get the data on when someone taps on a prayer category and which category that was?

We will because I'm now passing the nav params as properties on the screen tracking, which for prayer, contains the type of prayer list.

**Comment by richarddubay** [comment]
Oh, I thought we would need another `track` on the `PrayerTabView` for that. Is that not the case?

**Comment by redreceipt** [comment]
> Oh, I thought we would need another `track` on the `PrayerTabView` for that. Is that not the case?

I didn't think we'd want one for just clicking the buttons, but only when they start praying.

**Comment by richarddubay** [comment]
🤷‍♂️
Just going by the notes in the description of the task.

**Comment by redreceipt** [comment]
I wrote those notes :)

—
Michael

**Comment by richarddubay** [comment]
We'll just go with this then and if we need to change it we can change it.

**Review by richarddubay — APPROVED** [review]

**Pull request merged**